components/generate_embeddings/src/main.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import pandas as pd
+from fondant.component import PandasTransformComponent
+from langchain.embeddings import (
+    AlephAlphaAsymmetricSemanticEmbedding,
+    CohereEmbeddings,
+    HuggingFaceEmbeddings,
+    OpenAIEmbeddings,
+)
+from retry import retry
+from utils import to_env_vars
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateEmbeddingsComponent(PandasTransformComponent):
+    def __init__(
+        self,
+        *_,
+        model_provider: str,
+        model: str,
+        api_keys: dict,
+    ):
+        self.model_provider = model_provider
+        self.model = model
+
+        to_env_vars(api_keys)
```

Comment:
Could you extend the readme on how to use the different model providers? E.g. in how the api key environment variable should be named. I can assume that this differs from provider to provider.